src/Adapter/LegacyContext.php

```diff
@@ -228,17 +236,37 @@ public function getEmployeeLanguageIso()
     }
 
     /**
-     * Returns Currency set for the current employee.
+     * Returns Currency set for the current employee, you can force to return the default
+     * currency if the employee one is not set.
+     *
+     * @param bool $defaultFallback
+     *
+     * @return Currency
      */
-    public function getEmployeeCurrency()
+    public function getEmployeeCurrency($defaultFallback = false)
     {
-        static $employeeCurrency;
+        if (null === $this->employeeCurrency && $this->getContext()->currency) {
+            $this->employeeCurrency = $this->getContext()->currency->sign;
+        }
+        if (null === $this->employeeCurrency && $defaultFallback) {
+            return $this->getDefaultCurrency();
+        }
 
-        if (null === $employeeCurrency) {
-            $employeeCurrency = $this->getContext()->currency->sign;
+        return $this->employeeCurrency;
+    }
+
+    /**
+     * Returns default cCurrency set in Configuration
+     *
+     * @return Currency
+     */
+    public function getDefaultCurrency()
+    {
+        if (null === $this->defaultCurrency && ($id_currency = (int)Configuration::get('PS_CURRENCY_DEFAULT'))) {
```

Comment:
As far as I know, the configuration value of `PS_CURRENCY_DEFAULT` is always considered different zero, and does not need to be tested.

Reply:
ok I will simplify this method then

Reply:
done